packages/data/src/components/with-select/index.js

```diff
@@ -137,7 +148,13 @@ const withSelect = ( mapSelectToProps ) => createHigherOrderComponent( ( Wrapped
 		}
 
 		subscribe( registry ) {
-			this.unsubscribe = registry.subscribe( this.onStoreChange );
+			this.unsubscribe = registry.subscribe( () => {
```

Comment:
Given this is a hot code path, I'm inclined to wonder whether there's a way we can have this be prebound, rather than consider the condition of `this.props.isAsync` on each store update. In other words, can we assign `this.onStoreChange` to one which is wrapped with the `renderQueue.add` based on the `isAsync` prop?

Reply:
As a micro-optimization, and given it's not immediately obvious how it could be done to satisfaction, I think it ought to be deferred as a future optimization effort.